shared/build-tools/associate-types-in-packagejson.js

```diff
@@ -0,0 +1,38 @@
+const getFiles = require('node-recursive-directory');
```

Comment:
I have no idea whats going on in this file but I trust you do :)

Reply:
Dang, I should've put better comments in the top explaining the purpose. It's just looking for index.d.ts location and writing it back to each package.json so types are always correct instead of hardcoding them and hoping the type definitions structure stays the same (which they don't!)